spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/context/ContextStackAutoConfiguration.java

```diff
@@ -60,24 +65,29 @@
 	private AWSCredentialsProvider credentialsProvider;
 
 	@Bean
-	@ConditionalOnMissingBean(StackResourceRegistry.class)
-	public StackResourceRegistryFactoryBean stackResourceRegistryFactoryBean(
-			AmazonCloudFormation amazonCloudFormation) {
-
-		if (StringUtils.hasText(environment.getProperty("cloud.aws.stack.name"))) {
-			return new StackResourceRegistryFactoryBean(amazonCloudFormation,
-					new StaticStackNameProvider(
-							this.environment.getProperty("cloud.aws.stack.name")));
-		}
+	@ConditionalOnMissingBean
+	@ConditionalOnProperty("cloud.aws.stack.name")
+	public StackNameProvider staticStackNameProvider() {
```

Comment:
it is the expected behaviour that `manual` setting has a priority over the default `auto-detection` but I think we are getting the expected result due to the order of the beans. I would prefer to be explicit about it. In this case,

```
if (StringUtils.hasText(this.awsStackProperties.getName)) {
    return new StaticStackNameProvider(awsStackProperties.getName());
}
else {
    return new AutoDetectingStackNameProvider(amazonCloudFormation, this.amazonEC2);
}
```

Also, we can delete `auto` property. WDYT?

Reply:
Indeed we are getting correct behaviour because of the order of beans. Here the problem is that if we change it the way you suggested we are changing the behaviour of the method. I still do want to change how stack autoconfiguration works and potentially combine `auto` and `stackName` into single property but this can happen in 2.3.x. This bugfix is meant for 2.2.x

Reply:
I was thinking about it. Sounds good to me then :)